modules/rhsm_manifest.py

```diff
@@ -0,0 +1,264 @@
+    name: "katello.example.com"
+    username: "john-smith"
+    password: "changeme"
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 7
+    state: present
+    validate_certs: false
+
+- name: Ensure my manifest has 10 of one subs in it and export
+  rhsm_manifest:
+    uuid: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+    username: john-smith
+    password: changeme
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 10
+    state: present
+    validate_certs: false
+    path: /root/manifest.zip
+
+- name: Remove all of one subs from katello.example.com
+  rhsm_manifest:
+    name: katello.example.com
+    username: john-smith
+    password: changeme
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    state: absent
+    validate_certs: false
+
+'''
+
+RETURN = '''# '''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def fetch_portal(module, path, method, data={}, accept_header='application/json'):
+    url = module.params['portal'] + path
+    headers = {'accept': accept_header,
+               'content-type': 'application/json'}
+    resp, info = fetch_url(module, url, json.dumps(data), headers, method)
+    return resp, info
+    if resp is None:
+        try:
+            error = json.loads(info['body'])['displayMessage']
+        except:
+            error = info['msg']
+        raise Exception(error)
+    return resp, info
+
+
+def create_manifest(module):
+    path = "/subscription/consumers"
+    data = {'name': module.params['name'],
+            'type': "satellite",
+            'owner': module.params['rhsm_owner'],
+            'facts': {'distributor_version': 'sat-6.0',
+                      'system.certificate_version': '3.2'}}
+    resp, info = fetch_portal(module, path, 'POST', data)
+    return json.loads(resp.read())
+
+
+def get_or_create_manifest(module):
+    path = "/subscription/owners/%s/consumers?type=satellite" % (module.params['rhsm_owner'])
+    resp, info = fetch_portal(module, path, 'GET')
+    manifests = json.loads(resp.read())
+    if module.params['name']:
+        attr = 'name'
+    if module.params['uuid']:
+        attr = 'uuid'
+    manifest = filter(lambda m: m[attr] == module.params[attr], manifests)
+    if manifest:
+        return manifest[0], False
+    else:
+        return create_manifest(module), True
+
+
+def get_owner(module):
+    path = "/subscription/users/%s/owners" % (module.params['username'])
+    resp, info = fetch_portal(module, path, 'GET')
+    return json.loads(resp.read())[0]['key']
+
+
+def get_subs(module, manifest):
+    path = "/subscription/consumers/%s/entitlements" % (manifest['uuid'])
+    resp, info = fetch_portal(module, path, 'GET')
+    all_subs = json.loads(resp.read())
+    subs = filter(lambda s: s['pool']['id'] == module.params['pool_id'], all_subs)
```

Comment:
Note that in Python 3 filter has become a generator which means you can consume it only once. That may lead to very subtle breakage. I'd advise a list comprehension: ```[s for s in all_subs if s['pool']['id'] == module.params['pool_id']]``` or wrapping the above in ```list()```